zephyr/CMakeLists.txt

```diff
@@ -74,9 +85,55 @@ if(CONFIG_SOC_ESP32)
     ../components/driver/periph_ctrl.c
     ../components/log/log_noos.c
     ../components/log/log.c
+    adapter/src/common/dport_access.c
     adapter/src/stubs.c
     )
 
+  ## shared WIFI/BT resources
+  if (CONFIG_BT OR CONFIG_WIFI_ESP32)
+    zephyr_sources(
+      ../components/esp_wifi/src/phy_init.c
+      adapter/src/common/esp_system_api.c
+      )
+
+    zephyr_link_libraries(
+      net80211
+      core
+      pp
+      coexist
+      phy
+      rtc
+      ## esp-idf wifi libs refer gcc libs symbols, so linked in libgcc
+      gcc
+        -L${CMAKE_CURRENT_SOURCE_DIR}/../components/esp32
+        -L${CMAKE_CURRENT_SOURCE_DIR}/../components/esp_wifi/lib/esp32
+        -Wl,--defsym,phy_printf=printf
+        -Wl,--defsym,net80211_printf=printf
+        -Wl,--defsym,coexist_printf=printf
+      )
+
+  endif()
+
+  ## BT definitions
```

Comment:
I think this comment is unnecessary since the configuration option is self explanatory.